pre-docs/writing_yaml_plans.md

```diff
@@ -473,3 +473,134 @@ plan mymodule::yamlplan(
 }
 ```
 
+## YAML plan constructs which cannot be translated to Puppet plans
+
+This section points out some of the quirks and limitations associated with converting a plan expressed as YAML to a Puppet plan. In some cases it is impossible to accurately translate from YAML to Puppet, in others code that is generated from the conversion is syntactically correct but not idiomatic Puppet. 
```

Comment:
To avoid the run-on sentence feel, I'd break this long sentence into two:
> In some cases it is impossible to accurately translate from YAML to Puppet. In others, code that is generated from the conversion is syntactically correct but not idiomatic Puppet.